test/common/network/udp_listener_impl_test.cc

```diff
@@ -58,6 +61,15 @@ class UdpListenerImplTest : public ListenerImplTestBase {
                                      server_socket_->localAddress()->ip()->port());
   }
 
+  Address::Instance* getServerAnyAddress() {
+    if (version_ == Address::IpVersion::v4) {
+      return new Address::Ipv4Instance(Network::Test::getAnyAddressString(version_),
+                                       server_socket_->localAddress()->ip()->port());
+    }
+    return new Address::Ipv6Instance(Network::Test::getAnyAddressString(version_),
+                                     server_socket_->localAddress()->ip()->port());
```

Comment:
Ipv(4|6)Instance has a constructor to construct a address with "any ip" and a specified port.

```
/**
   * Construct from a port. The IPv4 address will be set to "any" and is suitable for binding
   * a port to any available address.
   */
explicit Ipv4Instance(uint32_t port);

/**
   * Construct from a port. The IPv6 address will be set to "any" and is suitable for binding
   * a port to any available address.
   */
explicit Ipv6Instance(uint32_t port);
```

Reply:
Thanks for pointing this out.
I had previously tried that implementation, but it did not give me the same "any ip" [::] but instead set the addr as [::1].
Hence, I am using the implementation as previously used here.

Reply:
Interesting. In that case, maybe file a issue for it, or just fix it in a separate PR?

Reply:
Where is this used?